src/wp-includes/blocks.php

```diff
@@ -2356,3 +2356,34 @@ function _wp_footnotes_force_filtered_html_on_import_filter( $arg ) {
 	}
 	return $arg;
 }
+
+/**
+ * Returns the active block variation for a given block based on its attributes.
+ *
+ * Variations are determined by their `isActive` property, which is an array of
+ * block attribute keys that are compared to the given block's attributes using
+ * a strict equality check.
+ *
+ * If no `isActive` property is defined, all `attributes` specified by a variation
+ * are compared to the given block's to determine if the variation is active.
+ *
+ * @param WP_Block_Type $block_type       Block Type.
```

Comment:
Should we change this to accept a block type _name_ instead to make it consistent with the client-side [`getActiveBlockVariation()`](https://developer.wordpress.org/block-editor/reference-guides/data/data-core-blocks/#getactiveblockvariation)? It will mostly mean some additional (and somewhat redundant) calls to `WP_Block_Type_Registry::get_instance()->get_registered()` to get the `WP_Block_Type` instance.

Reply:
Yeah, I was leaning that way initially, too. Now I've seen how similar a lot of this code here is to the client side (e.g. or `generated-classname.php`/`generated-class-name.js` or `getBlockDefaultClassName`/`wp_get_block_default_classname`) that I'm starting to reconsider 😅

Reply:
I would rather update the client side code to be more flexible. Most of these API operate on the string or object passed to represent a block type. However internally, it often requires converting the string to the block type object.